File: apps/desktop/src/ipc/screenshots.ts
import { invoke } from '@tauri-apps/api/core'

/** OCR engine name active on this platform, or `null` if text search is
 *  unavailable — the Screenshots settings pane's status line (the pane
 *  itself stays native; the search/grid view moved to `extensions/screenshots`
 *  in T29). */
export function screenshotOcrStatus(): Promise<string | null> {
  return invoke('screenshot_ocr_status')
}

File: src-tauri/src/api/screenshots.rs
/// Which OCR engine (if any) is available on this platform — drives the
/// Screenshots settings pane's status line.
#[tauri::command]
pub fn screenshot_ocr_status() -> Option<String> {
    crate::infrastructure::platform::ocr::available().then(|| crate::infrastructure::platform::ocr::engine_name().to_string())
}
